src/csv_module.f90

```diff
@@ -1217,7 +1217,7 @@ pure subroutine split(str,token,chunk_size,vals)
         call expand_vector(itokens,n_tokens,chunk_size,i+j)  ! save the token location
         if (i+len_token>len_str) exit  ! if the last bit of the string is a token
         j = j + i
-        temp = temp(i+len_token:len_str)  !remove previously scanned part of string
+        temp = trim(temp(i+len_token:len_str))  !remove previously scanned part of string
```

Comment:
Well, trim only removes trailing whitespace. So it would not remove delimiting whitespace.
The issue with temp = temp(start:end) is that some compilers on some platforms will not correctly reassign the new string. In my case when I apply the split function to the string "1.0,2.0,3.0,4.0,5.0" and follow the evolution of temp during the do loop i get:
- "2.0,3.0,4.0np.d"
- "3.0,4.0,4.0"
- "4.0?M-q"
To be honest, I don't really understand this behaviour. Please tell me if you can identify the core of the problem. In my case temp=trim(temp(start:end)) solves the problem.

I could check if I can observe the same behaviour on my linux machine&gfortran or with ifort?